WordPress/wordcamp.org
# Post Types: Add gutenberg support on speaker & organizer post types

Started in #303, this switches the metaboxes on Speaker and Organizer posts to use native gutenberg panels. It also refactors the sessions controls due to introducing a new hook, `usePostMeta`. Since these panels are introduced, this also restricts Speaker & Organizer to the gutenberg editor only.

### Screenshots

Organizer:
![organizer](https://user-images.githubusercontent.com/541093/83577530-6b7b9400-a502-11ea-9097-7a6cfeb9e526.png)

Speaker:
![speaker](https://user-images.githubusercontent.com/541093/83577532-6c142a80-a502-11ea-9370-d61748fb9048.png)

The usernames are validated when saving, so invalid usernames will be cleared from the input. On save, there's a hook that runs to fetch the user ID and fill in `_wcpt_user_id`. This runs for both organizers & speakers, since both use the same meta key.

### How to test the changes in this Pull Request:

1. Try adding or editing a speaker
2. Fill out the email & username
3. Save— the values should save correctly (check either by reloading, or using wp-cli's `post meta list`, etc)
4. Repeat for an organizer
5. Try with invalid usernames, they should not save (it would be a better UX for these to not fail silently, but this replicates the current flow)
